giskard/client/giskard_client.py

```diff
@@ -138,6 +138,9 @@ def __init__(self, url: str, key: str, hf_token: str = None):
 
         self._session.mount(url, adapter)
 
+        if self.key is None:
```

Comment:
Nitpicking here, but could you put this check earlier ? like line 133

It's good practice to validate as soon as you can, then do the logic